src/Interpolation/Zigzag.hpp

```diff
@@ -0,0 +1,121 @@
+namespace ippl {
+    namespace detail {
+        template <unsigned long ScatterPoint, unsigned long... Index, typename T, unsigned Dim,
+                  typename IndexType>
+        KOKKOS_INLINE_FUNCTION constexpr int ZigzagScatterToPoint(
+            const std::index_sequence<Index...>&,
+            const typename ippl::detail::ViewType<ippl::Vector<T, Dim>, Dim>::view_type& view,
+            const Vector<T, Dim>& wlo, const Vector<T, Dim>& whi,
+            const Vector<IndexType, Dim>& args, const Vector<T, Dim>& val, T scale,
+            const Vector<T, Dim>&, const NDIndex<Dim>, int, const Vector<T, Dim>&) {
+            bool isinbound = true;
+            ippl::Vector<T, Dim> depot =
+                scale * val * (interpolationWeight<ScatterPoint, Index>(wlo, whi) * ...);
+            ippl::Vector<IndexType, Dim> index3{interpolationIndex<ScatterPoint, Index>(args)...};
+            for (unsigned int d = 0; d < Dim; d++) {
+                isinbound &= (index3[d] < view.extent(d));
+            }
+            if (!isinbound) {
+                return 0;
+            }
+            for (unsigned int d = 0; d < Dim; d++) {
+                Kokkos::atomic_add(
+                    &(view(interpolationIndex<ScatterPoint, Index>(args)...)[d]),
+                    scale * val[d] * (interpolationWeight<ScatterPoint, Index>(wlo, whi) * ...));
+            }
+
+            return 0;
```

Comment:
The loop can be nested instead of having the early return. The return value itself is a dummy value anyway so it makes more sense to only return once.